### DISCUSSION TITLE
ScriptReference/Profiling.Memory.Experimental.MemoryProfiler.TakeSnapshot

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#75] [Notes]
# ScriptReference/Profiling.Memory.Experimental.MemoryProfiler.TakeSnapshot

https://docs.unity3d.com/ScriptReference/Profiling.Memory.Experimental.MemoryProfiler.TakeSnapshot.html

**Comment by vertxxyz** [comment]
`finishCallback` provides the path of the newly generated snapshot (null/empty if it failed) and a success boolean.
If a capture fails with little context, make sure you're writing to a location with permissions and disk space.
The default capture is `ManagedObjects | NativeObjects`, capturing less than this seems to break the Memory Profiler window.
